### ISSUE TITLE
Cancel post while media uploading

### ISSUE THREAD

**Issue by meremagee** [wordpress-mobile/WordPress-iOS-Editor#470]
We currently force users to wait until all media uploads are complete before they can cancel the post. This can take minutes on a slow connection if someone has added several images.

https://lookback.io/watch/Y5suZxN5THWMJwdC5

Let's consider allowing users to cancel the uploads in the same flow as canceling the post

**Comment by meremagee** [comment]
Suggested flows:

- new post > cancel post. regardless of whether media is uploading or not, show slide up alert: "you have unsaved changes. discard/save draft/keep editing." If user taps discard > cancel post and all uploads. If user taps save draft > throw alert to wait until imgs upload. If user taps keep editing, continue upload per usual.

**Comment by SergioEstevao** [comment]
Maybe you don't know this, but if you click on the media uploading.. message on the navbar it shows you an alert than allows you to cancel all ongoing uploads.

**Comment by meremagee** [comment]
Thanks @SergioEstevao - I'm aware of the current flow. I tagged this for @thianhlu for a UX review because it's not very intuitive and essentially removes an action that, I believe, should be an anchor/extremely consistent. When all else fails, the user should be able to hit `Cancel`, see an expected screen, and have the option to immediately exit the post. It is a lot to ask, especially of new users, to intuit that they should a) hit cancel b) get alert that they can't cancel c) hit uploading d) get alert/option to cancel upload e) hit cancel again f) see normal screen... without any UI hints.

Not urgent, but this feels broken usability-wise (not functionality-wise, since it works as we wrote it).
